internal/gcs/container.go

```diff
@@ -26,6 +26,10 @@ type Container struct {
 	notifyCh  chan struct{}
 	closeCh   chan struct{}
 	closeOnce sync.Once
+	// waitBlock is the channel used to wait for container shutdown or termination
+	waitBlock chan struct{}
+	// waitError indicates the container termination error if any
+	waitError error
```

Comment:
Can you add:
```go
// Should only be written to in [waitBackground]. 
// Safe to read after [waitBlock] is closed.
```

Reply:
This has been added in comments for WaitChannel() and WaitError() in internal/cow/cow.go